### ISSUE TITLE
Acknowledgement is not being propagated to parent Alarms

### ISSUE THREAD

**Issue by sfehlandt** [IntegratedAlarmSystem-Group/ias-webserver#43]
Currently, when all the children of and alarm are acknowledged, the parent alarm is not acknowledged.

Acknowledgement should work traversing the alarms tree upwards. This means, when acknowledging the last unacknowledged children of an Alarm, the alarm (the parent) should be acknowledged as well.

**Comment by sfehlandt** [comment]
Previously we experimented this issue randomly, that is, sometimes the parent alarm was acknowledged and sometimes not.

This can be explained by the fact that, when that happened, the Alarms that we used to test the acknowledgement were not in the CDB fixtures of the webserver, and hence, were not initialised when the webserver started. So when the alarm arrived from the core, it was added to the alarms_collection with its dependencies.

Up to this point, it should always work. However, if a parent alarm arrives to the webserver for the first time before any of its children has arrived, then the parent alarm is stored without dependencies. This is because, when we add an alarm to the collection, we check wether its dependencies are effectively Alarms or other-type IASIOS, which is done by checking if each of the dependencies exists or not in the collection. Since the parent is the first to arrive, its children are not in the collection, and hence the parent is stored as childrenless. This causes the parent_collection to not have the parenting relationships associated to that particular parent and its children, causing the (random) cases where the acknowledge did not work.
